src/main/scala/com/contxt/kinesis/ConsumerConfig.scala

```diff
@@ -0,0 +1,119 @@
+    )
+
+  def withInitialStreamPositionAtTimestamp(time: Date): ConsumerConfig =
+    this.copy(
+      initialPositionInStreamExtended = InitialPositionInStreamExtended.newInitialPositionAtTimestamp(time)
+    )
+
+  def withCoordinatorConfig(config: CoordinatorConfig): ConsumerConfig =
+    this.copy(coordinatorConfig = Some(config))
+
+  def withLeaseManagementConfig(config: LeaseManagementConfig): ConsumerConfig =
+    this.copy(leaseManagementConfig = Some(config))
+
+  def withMetricsConfig(config: MetricsConfig): ConsumerConfig = this.copy(metricsConfig = Some(config))
+
+  def withRetrievalConfig(config: RetrievalConfig): ConsumerConfig = this.copy(retrievalConfig = Some(config))
+
+  def withWorkerId(workerId: String): ConsumerConfig = this.copy(workerId = workerId)
+}
+
+object ConsumerConfig {
+  def apply(streamName: String, appName: String): ConsumerConfig = {
+    val kinesisClient = KinesisAsyncClient.builder.build()
+    val dynamoClient = DynamoDbAsyncClient.builder.build()
+    val cloudWatchClient = CloudWatchAsyncClient.builder.build()
+
+    withNames(streamName, appName)(kinesisClient, dynamoClient, cloudWatchClient)
+  }
+
+  def withNames(streamName: String, appName: String)
+               (implicit kac: KinesisAsyncClient, dac: DynamoDbAsyncClient, cwac: CloudWatchAsyncClient): ConsumerConfig =
+    ConsumerConfig(
+      streamName,
+      appName,
+      generateWorkerId(),
+      kac,
+      dac,
+      cwac
+    )
+
+  def fromConfig(config: Config)
+                (implicit
+                 kac: KinesisAsyncClient = null,
+                 dac: DynamoDbAsyncClient = null,
+                 cwac: CloudWatchAsyncClient = null
+                ): ConsumerConfig = {
+    def getOpt[A](key: String, lookup: String => A): Option[A] = if (config.hasPath(key)) Some(lookup(key)) else None
+
+    def getStringOpt(key: String): Option[String] = getOpt(key, config.getString)
+
+    val StreamPositionLatest = "latest"
+    val StreamPositionHorizon = "trim-horizon"
+    val StreamPositionTimestamp = "at-timestamp"
+
+    val streamName = config.getString("stream-name")
+    val name = config.getString("application-name")
+    val latestPos = InitialPositionInStreamExtended.newInitialPosition(InitialPositionInStream.LATEST)
+
+    val streamPosition = getStringOpt("position.initial")
+      .map {
+        case StreamPositionLatest => latestPos
+        case StreamPositionHorizon =>
+          InitialPositionInStreamExtended.newInitialPosition(
+            InitialPositionInStream.TRIM_HORIZON)
+        case StreamPositionTimestamp =>
+          InitialPositionInStreamExtended.newInitialPositionAtTimestamp(
+            DateFormat.getInstance().parse(config.getString("position.time")))
+      }
+      .getOrElse(latestPos)
+
+    val kinesisClient = Option(kac).getOrElse(KinesisClientUtil.createKinesisAsyncClient(KinesisAsyncClient.builder()))
```

Comment:
aws recommends to use [KinesisClientUtil](https://docs.aws.amazon.com/streams/latest/dev/kcl-migration.html) to create client